src/main/java/com/commercetools/sync/categories/CategorySync.java

```diff
@@ -144,7 +145,7 @@ public CategorySync(@Nonnull final CategorySyncOptions syncOptions) {
      */
     @Override
     protected CompletionStage<CategorySyncStatistics> processBatch(@Nonnull final List<CategoryDraft> categoryDrafts) {
-        final int numberOfNewDraftsToProcess = getNumberOfProcessedCategories(categoryDrafts);
+        final int numberOfNewDraftsToProcess = getNumberOfDraftsToProcess(categoryDrafts);
         referencesResolvedDrafts = new HashSet<>();
```

Comment:
1. why all the earlier initialized as concurrent maps/sets here are reinitialized and non-concurrent? (see categoryKeysWithResolvedParents)

2. Why they are initialized twice (in the field declaration here)?

3. Looks like some fields (at least `existingCategoryDrafts`, didn't review other) should be converted to local variables, since they are used only in context of `processBatch()` function.

Reply:
1. That is actually my mistake, thanks for noticing it!
2. They are initialized twice because they need to be emptied on every batch processing.
3. They have global access, they are not only used here, but they are used by several methods.